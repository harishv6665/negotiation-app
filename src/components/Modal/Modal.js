import React from "react";
import styles from "./Modal.module.css";
import Button from "../Button/Button";

const Modal = ({ title, temperature, children, onClose }) => (
    <div className={styles.overlay}>
        <div className={styles.modal}>
            <h4 className={styles.modal__title}>{title}</h4>
            <div className={styles.modal__body}>{children}</div>
            <footer className={styles.modal__footer}>
                <Button onClick={onClose} title="Ok" />
                <p className={styles.modal__footer__msg}>Current temperature in london: <strong>{temperature}</strong></p>
            </footer>
        </div>
    </div>
)

export default Modal;
