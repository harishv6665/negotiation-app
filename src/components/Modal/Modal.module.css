.overlay {
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background-color: rgba(0, 0, 0, 0.35);
    z-index: 9999;
}

.modal {
    position: absolute;
    max-width: 550px;
    width: 100%;
    top: 50%;
    left: 50%;
    background-color: #fff;
    transform: translate(-50%, -50%);
    padding: 30px 50px 54px;
    border-radius: 5px;
    overflow: hidden;
}

.modal__title {
    font-size: 46px;
    text-align: center;
    margin-bottom: 40px;
    font-weight: 400;
}

.modal__body {
    margin-bottom: 40px;
}

.modal__footer {

}

.modal__footer__msg {
    position: fixed;
    bottom: 0;
    left: 0;
    right: 0;
    font-size: 14px;
    height: 26px;
    background: #607D8B;
    color: #fff;
    display: flex;
    align-items: center;
    justify-content: center;
}

@media (max-width: 575px) {
    .modal {
        max-width: initial;
        width: calc(100% - 40px);
        min-width: 320px;
    }
}
