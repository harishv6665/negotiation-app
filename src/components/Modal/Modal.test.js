import React from 'react';
import { shallow, mount } from 'enzyme';
import Modal from "./Modal";

describe('Modal tests', () => {
    const MODAL_TITLE = "Success!";
    const MODAL_TEMPERATURE = "208";
    const MODAL_CONTENT = "Modal content text";

    it('should be defined', () => {
        expect(Modal).toBeDefined();
    });

    it('should match to snapshot', () => {
        const wrapper = shallow(
            <Modal
                title={MODAL_TITLE}
                temperature={MODAL_TEMPERATURE}
            />
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('should match the DOM structure', () => {
        const wrapper = mount(
            <Modal
                title={MODAL_TITLE}
                temperature={MODAL_TEMPERATURE}
            >
                <p className="content">{MODAL_CONTENT}</p>
            </Modal>
        );
        expect(wrapper.find('h4').text()).toEqual(MODAL_TITLE);
        expect(wrapper.find('.content').text()).toEqual(MODAL_CONTENT);
        expect(wrapper.find('footer p strong').text()).toEqual(`${MODAL_TEMPERATURE}Â°C`);
        expect(wrapper.find('footer button').text()).toEqual("Done");
    });

    it('should simulate button click', () => {
        const mockFn = jest.fn();
        const wrapper = mount(
            <Modal
                title={MODAL_TITLE}
                temperature={MODAL_TEMPERATURE}
                onClose={mockFn}
            />
        );
        wrapper.find('footer button').simulate('click');
        expect(mockFn).toHaveBeenCalled();
    });

});
