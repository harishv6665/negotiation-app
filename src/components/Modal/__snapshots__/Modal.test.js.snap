// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal tests should match to snapshot 1`] = `ShallowWrapper {}`;
